tests/components/watergate/test_config_flow.py

```diff
@@ -0,0 +1,173 @@
+        assert result["type"] == data_entry_flow.FlowResultType.CREATE_ENTRY
+        assert result["title"] == "Test Device"
+        assert result["data"] == {
+            CONF_NAME: "Test Device",
+            CONF_IP_ADDRESS: "192.168.1.100",
+        }
+
+
+@pytest.mark.asyncio
+async def test_step_reconfigure_form(hass: HomeAssistant) -> None:
+    """Test displaying the reconfigure form with default values."""
+    # Initialize the config flow
+    flow = WatergateConfigFlow()
+    flow.hass = hass
+
+    # Mock an existing ConfigEntry to use for reconfigure
+    current_entry = MockConfigEntry(
+        version=1,
+        domain=DOMAIN,
+        title="Existing Device",
+        data={
+            CONF_NAME: "Existing Device",
+            CONF_IP_ADDRESS: "192.168.1.101",
+        },
+        source="user",
+        entry_id="12345",
+    )
+
+    with patch.object(flow, "_get_reconfigure_entry", return_value=current_entry):
+        result = await flow.async_step_reconfigure(user_input=None)
+
+        # Check that the form is returned with default values
+        assert result["type"] == data_entry_flow.FlowResultType.FORM
+        assert result["step_id"] == "reconfigure"
+
+
+@pytest.mark.asyncio
+async def test_step_reconfigure_update_entry(hass: HomeAssistant) -> None:
+    """Test updating an existing entry from reconfigure user input."""
+    # Initialize the config flow
+    flow = WatergateConfigFlow()
+    flow.hass = hass
+
+    # Mock an existing entry to use for reconfigure
+    current_entry = MockConfigEntry(
+        entry_id="12345",
+        data={
+            CONF_NAME: "Existing Device",
+            CONF_IP_ADDRESS: "192.168.1.101",
+        },
+    )
+
+    # Define mock user input for updating
+    new_user_input = {
+        CONF_NAME: "Updated Device",
+        CONF_IP_ADDRESS: "192.168.1.102",
+    }
+
+    # Patch `_get_reconfigure_entry` and `async_update_reload_and_abort`
+    with (
+        patch.object(flow, "_get_reconfigure_entry", return_value=current_entry),
+        patch.object(flow, "async_update_reload_and_abort") as mock_update_abort,
+        patch(
+            "homeassistant.components.watergate.config_flow.WatergateLocalApiClient"
+        ) as mock_client,
+    ):
+        mock_client_instance = mock_client.return_value
+        mock_client_instance.async_get_device_state = AsyncMock(
+            return_value=DeviceState(
+                "open",
+                "on",
+                True,
+                True,
+                "battery",
+                "1.0.0",
+                100,
+                {"volume": 1.2, "duration": 100},
+            )
+        )
+        await flow.async_step_reconfigure(user_input=new_user_input)
+
+    mock_update_abort.assert_called_once_with(
+        current_entry,
+        data_updates={
+            CONF_NAME: "Updated Device",
+            CONF_IP_ADDRESS: "192.168.1.102",
+        },
+    )
```

Comment:
you'll now also need a test for the error case during validation (you'll need 100% coverage for the configflow)